instrumentation/graphql_batch/lib/opentelemetry/instrumentation/graphql/batch/patches/loader.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module GraphQL
+      module Batch
+        module Patches
+          # Module to prepend to GraphQL::Batch::Loader for instrumentation
+          module Loader
+            def around_perform
+              tracer.in_span(span_name, attributes: attributes) do
+                super
+              end
+            end
+
+            private
+
+            def tracer
+              GraphQL::Batch::Instrumentation.instance.tracer
+            end
+
+            def span_name
+              loader_name.split('::').last
```

Comment:
Do we care about cases where the last name is not unique, e.g. `Blah::Blah::Redis::Loader` and `Yuck::Yuck::ActiveRecord::Loader`?

Reply:
I think that's ok, , the full path is available as an attribute.